go/libraries/doltcore/doltdb/commit_hooks.go

```diff
@@ -93,6 +95,61 @@ func pushDataset(ctx context.Context, destDB, srcDB datas.Database, tempTableDir
 	return err
 }
 
+type PushArg struct {
+	ds   datas.Dataset
+	db   datas.Database
+	hash hash.Hash
+}
+
+type AsyncPushOnWriteHook struct {
+	out io.Writer
+	ch  chan PushArg
+}
+
+const asyncPushBufferSize = 20
```

Comment:
I think you may want to bump this up much higher.  At some point Dolt will be running with many simultaneous connections each of which may be writing.  Don't want to block when writing to the channel in this case.

Reply:
2048?